va/va.go

```diff
@@ -524,3 +540,30 @@ func (va VAImpl) fetchHTTP(identifier string, token string) ([]byte, string, *ac
 
 	return body, url.String(), nil
 }
+
+//this reverseaddr funntion is from net/dnsclient.go form golang
+func reverseaddr(addr string) string {
+	const hexDigit = "0123456789abcdef"
+	ip := net.ParseIP(addr)
+	if ip == nil {
+		return ""
+	}
+	// Apperently IP type in net package saves all ip in ipv6 formant, from biggest byte to smallest. we need last 4 bytes, so ip[15] to ip[12]
+	if ip.To4() != nil {
+		return strconv.Itoa(int(ip[15])) + "." + strconv.Itoa(int(ip[14])) + "." + strconv.Itoa(int(ip[13])) + "." + strconv.Itoa(int(ip[12])) + ".in-addr.arpa."
+	}
+	// Must be IPv6
+	buf := make([]byte, 0, len(ip)*4+len("ip6.arpa."))
```

Comment:
Rather than doing buffer based string construction like this it'd be somewhat simpler to just use a `[]string` and append each two quads to it in the loop below then do the final construction using `fmt.Sprintf("%s.ip6.arpa.", strings.Join(quads, "."))`.

Reply:
I can't find good name for this []string. it's single byte, but upper and lower is reversed.